src/types/dispenser-client.ts

```diff
@@ -0,0 +1,179 @@
+ * either directly as a parameter or through an `ALGOKIT_DISPENSER_ACCESS_TOKEN` environment variable. If neither is provided, an error is thrown.
+ *
+ * The request timeout can be provided as a parameter. If not provided, a default value is used.
+ *
+ * @property {string} authToken - The authentication token used for API requests.
+ * @property {number} requestTimeout - The timeout for API requests, in seconds.
+ *
+ * @method fund - Sends a funding request to the dispenser API to fund the specified address with the given amount of Algo.
+ * @method refund - Sends a refund request to the dispenser API for the specified refundTxnId.
+ * @method limit - Sends a request to the dispenser API to get the funding limit for the Algo asset.
+ *
+ * @example
+ * ```typescript
+ * const client = new DispenserApiTestnetClient({ authToken: 'your_auth_token', requestTimeout: 30 });
+ * const fundResponse = await client.fund('your_address', 100);
+ * const limitResponse = await client.getLimit();
+ * await client.refund('your_transaction_id');
+ * ```
+ *
+ * @throws {Error} If neither the environment variable 'ALGOKIT_DISPENSER_ACCESS_TOKEN' nor the authToken parameter were provided.
+ */
+export class DispenserApiTestnetClient {
+  private _authToken: string
+  private _requestTimeout: number
+
+  constructor(params: DispenserApiTestnetClientParams | null) {
+    const authTokenFromEnv = process.env[dispenserAccessTokenKey]
+
+    if (params?.authToken) {
+      this._authToken = params.authToken
+    } else if (authTokenFromEnv) {
+      this._authToken = authTokenFromEnv
+    } else {
+      throw new Error(
+        `Can't init AlgoKit TestNet Dispenser API client because neither environment variable ${dispenserAccessTokenKey} or the authToken were provided.`,
+      )
+    }
+
+    this._requestTimeout = params?.requestTimeout || dispenserRequestTimeout
+  }
+
+  get authToken(): string {
+    return this._authToken
+  }
+
+  get requestTimeout(): number {
+    return this._requestTimeout
+  }
+
+  /**
+   * Processes a dispenser API request.
+   *
+   * @param authToken - The authentication token.
+   * @param urlSuffix - The URL suffix for the API request.
+   * @param body - The request body.
+   * @param method - The HTTP method.
+   *
+   * @returns The API response.
+   */
+  private async processDispenserRequest(
+    authToken: string,
+    urlSuffix: string,
+    body: Record<string, string | number> | null = null,
+    method = 'POST',
+  ): Promise<Response> {
+    const headers = { Authorization: `Bearer ${authToken}` }
+
+    const requestArgs: RequestInit = {
+      method: method,
+      headers: headers,
+      signal: AbortSignal.timeout(this.requestTimeout * 1000),
+    }
+
+    if (body) {
+      requestArgs.body = JSON.stringify(body)
+    }
+
+    const response = await fetch(`${baseUrl}/${urlSuffix}`, requestArgs)
```

Comment:
Are there any issues with relying on global fetch? We have minimum of node 18 don't we? Is global fetch available in node 18?

Reply:
@robdmoore didn't notice that we use croos-fetch in urltokenbasehttpclient class we already have in utils. Will open a follow up chore pr to address this